tests/testsuite/future_incompat_report.rs

```diff
@@ -0,0 +1,151 @@
+        .run();
+}
+
+#[cargo_test]
+fn gate_future_incompat_report() {
+    let p = project()
+        .file("Cargo.toml", &basic_manifest("foo", "0.0.0"))
+        .file("src/main.rs", "fn main() { [true].into_iter(); }")
+        .build();
+
+    p.cargo("build --future-incompat-report")
+        .with_stderr_contains(
+            "error: Usage of `--future-incompat-report` requires `-Z future-incompat-report`",
+        )
+        .with_status(101)
+        .run();
+
+    p.cargo("describe-future-incompatibilities --id foo")
+        .with_stderr_contains(
+            "error: `cargo describe-future-incompatibilities` can only be used on the nightly channel with `-Z future-incompat-report`"
+        )
+        .with_status(101)
+        .run();
+}
+
+#[cargo_test]
+fn test_single_crate() {
+    if !is_nightly() {
+        return;
+    }
+
+    let p = project()
+        .file("Cargo.toml", &basic_manifest("foo", "0.0.0"))
+        .file("src/main.rs", "fn main() { [true].into_iter(); }")
+        .build();
+
+    for command in &["build", "check", "rustc"] {
+        p.cargo(&format!("{} -Z future-incompat-report", command))
+            .masquerade_as_nightly_cargo()
+            .with_stderr_contains("  = note: `#[warn(array_into_iter)]` on by default")
+            .with_stderr_contains("warning: the following crates contain code that will be rejected by a future version of Rust: `foo` v0.0.0")
+            .with_stderr_does_not_contain("The crate `foo` v0.0.0 currently triggers the following future incompatibility lints:")
+            .run();
+
+        p.cargo(&format!("{} -Z future-incompat-report --future-incompat-report", command))
+            .masquerade_as_nightly_cargo()
+            .with_stderr_contains("  = note: `#[warn(array_into_iter)]` on by default")
+            .with_stderr_contains("warning: the following crates contain code that will be rejected by a future version of Rust: `foo` v0.0.0")
+            .with_stderr_contains("The crate `foo` v0.0.0 currently triggers the following future incompatibility lints:")
+            .run();
+    }
+}
+
+#[cargo_test]
+fn test_multi_crate() {
+    if !is_nightly() {
+        return;
+    }
+
+    Package::new("first-dep", "0.0.1")
+        .file("src/lib.rs", "fn foo() { [25].into_iter(); }")
+        .publish();
+    Package::new("second-dep", "0.0.2")
+        .file("src/lib.rs", "fn foo() { ['a'].into_iter(); }")
+        .publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.0"
+
+                [dependencies]
+                first-dep = "*"
+                second-dep = "*"
+              "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build();
+
+    for command in &["build", "check", "rustc"] {
+        p.cargo(&format!("{} -Z future-incompat-report", command))
```

Comment:
Instead of `format!` you can do `.cargo(command).arg("-Zfuture-...")`

Reply:
It seems like this was marked resolved, but the code looks the same.